### PR TITLE
Repurpose package

### PR DESCRIPTION
**Pull request by barweiss**
The new package provides testing utilities for the automatically generated Seekret Python tests

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -10,23 +10,23 @@ def read_readme():
...
-        "Programming Language :: Python :: 3",
-        "License :: OSI Approved :: MIT License",
-        "Operating System :: OS Independent"
-    ],
-    project_urls={
-        'Bug Tracker': 'https://github.com/seek-ret/tests-rtl/issues'
-    },
-    url='https://github.com/seek-ret/tests-rtl')
+setup(name='seekret.apitest',
+      version='0.2.0',
```
Comment (asdftamir) [inline setup.py]: bump version DAHUF

File: seekret/apitest/auth/headers.py [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
-    def on_test_start(self, test_data: Box, variables: Box, auth_data: Box):
-        variables.merge_update(
-            {'seekret-runtime': {
-                'v1': {
-                    'auth': {
-                        'headers': auth_data
-                    }
-                }
-            }})
+@auth_method_factory(type_name='headers')
```
Comment (asdftamir) [inline seekret/apitest/auth/headers.py]: header*

File: seekret/apitest/context/context.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+import requests
+from requests.structures import CaseInsensitiveDict
+
+from seekret.apitest.auth import create_auth
+from seekret.apitest.context.response import ResponseWrapper
+from seekret.apitest.runprofile import RunProfile
+
+
+def resolve_path_params(path: str, path_params: dict[str, Any]):
```
Comment (asdftamir) [inline seekret/apitest/context/context.py:16]: this should fail if some parameters were not specified/ too many parameters were specified

File: seekret/apitest/context/context.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        :param cookies: Cookies to send in the request.
+        :param user: The requesting user. The available users are defined in the run profile.
+                     Use `None` to send an unauthenticated request.
+        :return: Wrapped response object.
+        """
+
+        path = resolve_path_params(path, path_params)
+        return ResponseWrapper(
+            requests.request(method=method,
+                             url='/'.join([
```
Comment (asdftamir) [inline seekret/apitest/context/context.py]: `urllib.parse.urljoin`

**Review by asdftamir — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (barweiss) [reply setup.py]: Not sure it should be a part of this PR. But, I'll add a branch for the whole version so we can still fix stuff in `main`

File: seekret/apitest/auth/headers.py (hunk 2 above)
Comment (barweiss) [reply seekret/apitest/auth/headers.py]: Whoops, fixed

File: seekret/apitest/context/context.py (hunk 3 above)
Comment (barweiss) [reply seekret/apitest/context/context.py:16]: Fixed

File: seekret/apitest/context/context.py (hunk 4 above)
Comment (barweiss) [reply seekret/apitest/context/context.py]: Fixed

File: seekret/apitest/context/context.py [hunk 5]
```diff
@@ -100,12 +110,12 @@ def request(self,
         """
 
         path = resolve_path_params(path, path_params)
+
+        # Strip "/" at the start of the path to avoid "//" replacing the host part.
+        url = urllib.parse.urljoin(self.run_profile.target_server, path.lstrip('/'))
```
Comment (asdftamir) [inline seekret/apitest/context/context.py]: I think urljoin takes care of extra "/"

**Review by asdftamir — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (barweiss) [reply setup.py]: Bumped

File: seekret/apitest/context/context.py (hunk 5 above)
Comment (barweiss) [reply seekret/apitest/context/context.py]: Discussed

File: seekret/apitest/context/response.py [hunk 6]
```diff
@@ -36,3 +39,12 @@ def search(self, expression: str):
             'json': self.json(),
             'headers': self.headers
         })
+
+    def assert_schema(self, schema):
+        try:
+            pykwalify.core.Core(
+                source_data=self.json(),
+                schema_file_obj=io.StringIO(schema)).validate()
```
Comment (asdftamir) [inline seekret/apitest/context/response.py:47]: does this need some flag to represent its a yaml?

**Review by asdftamir — CHANGES_REQUESTED** [review]

**Review by asdftamir — APPROVED** [review]

**Pull request merged**